[DOCFIX] Add table docs

ZacBlanco opened the pull request.

ZacBlanco commented:
first draft, feel free to tear it apart

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/6191/
Test PASSed.

yuzhu commented:
@ZacBlanco thanks for writing the doc! 
a couple of high level comments: 
1. maybe add some examples so that the user can verify that they have successfully setup alluxio {{feature_name}} for presto.  Users like confirmations of success. things like `describe table`, and `select * from table_name` 
2.  do we want to make sure the user understands that the cache is not automatically synced at this point? i.e. there is no `msck repair table` equivalent.

apc999 commented on docs/en/compute/table.md:
```diff
@@ -0,0 +1,224 @@
+By default, presto will now retrieve database and table information from Alluxio's {{feature_name}}
+when executing any queries.
+
+
```
remove extra empty lines here

ZacBlanco replied:
fixed

apc999 commented on docs/en/compute/table.md:
```diff
@@ -0,0 +1,224 @@
+They store information about different database catalogs, tables, storage formats, data
+location, and more.
+{{feature_name}} is designed to make it simple and straightforward to retrieve and serve structured table
+metadata to these SQL engines.
```
can we specify a few bullets in why we want to introduce another catalog service, rather than playing with existing HMS? I can see benefits  for users like easy integration for Presto (or potentially other engines)  & Alluxio as no modification to HMS required, allowing more Alluxio-level optimizations (e.g. fwd reference transformation)?

ZacBlanco replied:
the only real advantage I know of right now is for transformations...

@gpang @yuzhu can you guys help come up with a list of reasons to use the alluxio table service?

apc999 commented on docs/en/compute/table.md:
```diff
@@ -0,0 +1,224 @@
+{{feature_name}} is designed in a way very similar to the normal Alluxio filesystem.
+The service itself is not responsible for retaining all data, but is rather a caching service for
+metadata that originates in another location (i.e. MySQL, Hive).
+These are called **UDBs**.
```
shall we call out that UDB stands for (Under Data Base)?

ZacBlanco replied:
updated

yuzhu commented on docs/en/compute/table.md:
````diff
@@ -0,0 +1,224 @@
+```console
+$ ${ALLUXIO_HOME}/bin/alluxio table attachdb alluxio_db hive \
+    --option udb-hive.database-name=default \
+    --option udb-hive.metastore.uris=thrift://metastore_host:9083
````
i think the option has been changed to `hive.hive.metastore.uris` ?

ZacBlanco replied:
updated

dborkar commented on docs/en/compute/table.md:
```diff
@@ -0,0 +1,224 @@
+Running `alluxio table ls` afterwards will not display the database any more.
+Continue to the next section to see how to use {{feature_name}} with presto
+
+## Data Transformations
```
I'd recommend taking Data Transformation out of the Catalog Service docs. It should live independent of the catalog and include information about how the catalog knows which target data version to use.

ZacBlanco commented:
@yuzhu I addressed the two comments you made:

1. Added example queries to test
2. Added a note on attaching databases about syncing database/table metadata.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/6227/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/6344/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/6355/
Test PASSed.

gpang commented on docs/_config.yml:
```diff
@@ -78,6 +79,9 @@ categoryInfo:
     cn: APIs
+  Structured Data:
+    en: Structured Data
+    cn: œ∑´®†¥¨ˆøπ“‘åß∂ƒ©˙∆˚¬…æ≈ç√∫˜µ≤≥÷
```
this looks wrong?

ZacBlanco replied:
fixed

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/6401/
Test PASSed.

ZacBlanco commented:
alluxio-bot, merge this please

ZacBlanco commented:
alluxio-bot, cherry-pick this to branch-2.1 please
